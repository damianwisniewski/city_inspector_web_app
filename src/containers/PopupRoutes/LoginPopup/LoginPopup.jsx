import React, { Component } from 'react'
// import PropTypes from 'prop-types'

import { Form, Input } from '../../../components/FormComponents'
import Button from '../../../components/CommonComponents/Button/Button'

import './LoginPopup.scss'
// import Status from '../../../components/CommonComponents/Status/Status'

class LoginPopup extends Component {
	state = {
		forgottenPassword: false,
		loginFailed: false,
	}

	handleForgottenPassword = () => {
		this.setState({
			forgottenPassword: true,
		})
	}

	render() {
		const { closePopup } = this.props
		const { forgottenPassword } = this.state

		if (forgottenPassword) {
			return (
				<div id='LoginPopup' className='login'>
					<h2 className='login__header'>Resetowanie hasła</h2>
					<p className='login__description'>
						Podaj adres e-mail na który zarejestrowane jest konto
					</p>
					<Form>
						<Input
							id='email-field'
							label='e-mail'
							placeholder='Wpisz tutaj swój email...'
							floatingLabel
							type='email'
							autocomplete='email'
							pattern='.+@.+'
							required
						/>
						<div className='login__buttons-wrapper'>
							<Button id='inform-close-button' color='white' onClick={closePopup}>
								Anuluj
							</Button>
							<Button id='submit-button' type='submit' color='blue'>
								Resetuj
							</Button>
						</div>
					</Form>
				</div>
			)
		} else {
			return (
				<div id='LoginPopup' className='login'>
					<h2 className='login__header'>Logowanie</h2>
					{/* <Status id='login-status' type='error' message='Niepoprawny email lub hasło' /> */}
					<p className='login__description'>Wpisz następujące dane aby się zalogować</p>
					<Form>
						<Input
							id='email-field'
							label='e-mail'
							placeholder='Wpisz tutaj swój email...'
							floatingLabel
							type='email'
							autocomplete='username'
							pattern='.+@.+'
							required
						/>
						<Input
							id='password-field'
							label='hasło'
							placeholder='Podaj hasło...'
							floatingLabel
							type='password'
							autocomplete='current-password'
							required
						/>
						<Button
							id='forgotten-pass-button'
							type='text'
							color='blue'
							onClick={this.handleForgottenPassword}
						>
							Zapomniałeś hasła?
						</Button>
						<div className='login__buttons-wrapper'>
							<Button id='inform-close-button' color='white' onClick={closePopup}>
								Anuluj
							</Button>
							<Button id='submit-button' type='submit' color='blue'>
								Zaloguj
							</Button>
						</div>
					</Form>
				</div>
			)
		}
	}
}

// LoginPopup.propTypes = {}

export default LoginPopup
