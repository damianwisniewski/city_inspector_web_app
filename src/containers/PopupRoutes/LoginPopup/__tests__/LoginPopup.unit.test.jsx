import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import LoginPopup from '../LoginPopup'

configure({ adapter: new Adapter() })

describe('<Loader />', () => {
	let LoginPopupElement

	test('should render correctly LoginPopup', () => {
		LoginPopupElement = shallow(<LoginPopup />)
		expect(toJson(LoginPopupElement)).toMatchSnapshot()
	})

	test('should render correctly LoginPopup with remember password for state forgottenPassword true', () => {
		LoginPopupElement = shallow(<LoginPopup />)
		LoginPopupElement.setState({ forgottenPassword: true })
		expect(toJson(LoginPopupElement)).toMatchSnapshot()
	})
})
