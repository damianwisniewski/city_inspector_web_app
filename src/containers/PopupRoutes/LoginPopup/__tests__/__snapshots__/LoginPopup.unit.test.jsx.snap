// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render correctly LoginPopup 1`] = `
<div
  className="login"
  id="LoginPopup"
>
  <h2
    className="login__header"
  >
    Logowanie
  </h2>
  <p
    className="login__description"
  >
    Wpisz następujące dane aby się zalogować
  </p>
  <Form>
    <Input
      autocomplete="username"
      floatingLabel={true}
      id="email-field"
      label="e-mail"
      pattern=".+@.+"
      placeholder="Wpisz tutaj swój email..."
      required={true}
      type="email"
    />
    <Input
      autocomplete="current-password"
      floatingLabel={true}
      id="password-field"
      label="hasło"
      placeholder="Podaj hasło..."
      required={true}
      type="password"
    />
    <Button
      className=""
      color="blue"
      id="forgotten-pass-button"
      onClick={[Function]}
      type="text"
    >
      Zapomniałeś hasła?
    </Button>
    <div
      className="login__buttons-wrapper"
    >
      <Button
        className=""
        color="white"
        id="inform-close-button"
        onClick={[Function]}
        type="button"
      >
        Anuluj
      </Button>
      <Button
        className=""
        color="blue"
        id="submit-button"
        onClick={[Function]}
        type="submit"
      >
        Zaloguj
      </Button>
    </div>
  </Form>
</div>
`;

exports[`<Loader /> should render correctly LoginPopup with remember password for state forgottenPassword true 1`] = `
<div
  className="login"
  id="LoginPopup"
>
  <h2
    className="login__header"
  >
    Resetowanie hasła
  </h2>
  <p
    className="login__description"
  >
    Podaj adres e-mail na który zarejestrowane jest konto
  </p>
  <Form>
    <Input
      autocomplete="email"
      floatingLabel={true}
      id="email-field"
      label="e-mail"
      pattern=".+@.+"
      placeholder="Wpisz tutaj swój email..."
      required={true}
      type="email"
    />
    <div
      className="login__buttons-wrapper"
    >
      <Button
        className=""
        color="white"
        id="inform-close-button"
        onClick={[Function]}
        type="button"
      >
        Anuluj
      </Button>
      <Button
        className=""
        color="blue"
        id="submit-button"
        onClick={[Function]}
        type="submit"
      >
        Resetuj
      </Button>
    </div>
  </Form>
</div>
`;
